webpack/scenes/Subscriptions/Manifest/SCAContainer.js

```diff
@@ -0,0 +1,61 @@
+import React, { useState, useRef } from 'react';
+import { translate as __ } from 'foremanReact/common/I18n';
+import PropTypes from 'prop-types';
+import SimpleContentAccess from './SimpleContentAccess';
+
+const SCAContainer = (props) => {
+  const {
+    isSimpleContentAccessEnabled,
+  } = props;
+  const [enabled, setEnabled] = useState(isSimpleContentAccessEnabled);
+  const hiddenRailsInput = useRef(document.querySelector('#content_access_mode'));
+
+  const infoMsgOverride = enabledVal => (enabledVal !== isSimpleContentAccessEnabled) || undefined;
+  const getContentAccessModeValue = (newValue) => {
```

Comment:
Observations
- I see that `infoMsgOVerride` changes the messaging on screen. It seems a little strange to mention that there's a change pending. We don't do that for any of the other form inputs, why this one? IMO it just adds more complexity in this component.
- We should avoid having the UI layer know about 'content access mode' and the related values. That really is a Candlepin concept and should be kept on the backend

Suggestions:
- rename the hidden input field to 'simple_content_access' and have its default value be @org.simple_content_access?
- implies the hidden field is now only true/false
- on the backend, if we **detect a change** between the current @org.simple_content_access? and the form value, perform the right action

Reply:
> We don't do that for any of the other form inputs, why this one? 

There were 2 ways I was thinking about consistency here: (1) the form element should behave the same way as all other form elements on the page (we already talked about this one), and (2) this same toggle is used both here and in the Manage Manifest modal on the Subscriptions page.  On the Subscriptions page, the change executes immediately, so users might also expect it to execute immediately here.  That's why I added the `infoMsgOverride` dynamic text.

>  the hidden field is now only true/false

The `value` attribute of the [`<input>`](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/hidden) HTML element is a [`DOMString`](https://developer.mozilla.org/en-US/docs/Web/API/DOMString), so I don't think trying to store a boolean value there would work well.

That said, I don't have any strong feelings about the name or text value of the attribute.  We just need to read the string from that value attribute and send it as a param to Rails.  If we want to call it `simple_content_access` and then have the value be something like `enable` or `disable`, I'd be fine with that.

> on the backend, if we detect a change between the current @org.simple_content_access? and the form value, perform the right action

It's doing this right now, but on the front end.  If we were to move this logic to the back end, it would remove the ability to have the `infoMsgOverride`.

Reply:
I just realized that using a switch here goes against [this Patternfly guideline](https://www.patternfly.org/v4/documentation/core/components/switch#overview):

_Use a switch when your user needs to perform instantaneous actions without confirmation._

_Use checkbox if your user has to perform additional steps for changes to become effective._

Going to look into changing this to a dumb checkbox.  This may remove the need for a React component entirely.

Reply:
Using a checkbox really would make this a lot simpler considering the patternfly guidance around when to use the switch and how it should behave (real time, instantaneous action triggering). +1 from me.

Reply:
> It's doing this right now, but on the front end. If we were to move this logic to the back end, it would remove the ability to have the `infoMsgOverride`.

I perceive it this way: your impl on the front end is to enable dynamic messaging, but it has a side-effect of not triggering a manifest action on the backend if the user didn't touch the toggle since it will send 'unchanged' as the value. This tightly couples the front and back end. If someone comes along later and changes the form (intentionally or not) it would be easy for the `update` method to start triggering the SCA toggle action on each `update`. No matter what the front end is doing, it's safer to have that defensive check in the controller to prevent future bugs.